common/src/services/vaultTimeout.service.ts

```diff
@@ -50,110 +50,96 @@ export class VaultTimeoutService implements VaultTimeoutServiceAbstraction {
     }
 
     // Keys aren't stored for a device that is locked or logged out.
-    async isLocked(): Promise<boolean> {
-        // Handle never lock startup situation
-        if (await this.cryptoService.hasKeyStored('auto') && !this.everBeenUnlocked) {
-            await this.cryptoService.getKey('auto');
+    async isLocked(userId?: string): Promise<boolean> {
+        const neverLock = await this.cryptoService.hasKeyStored(KeySuffixOptions.Auto, userId) &&
+            !(await this.stateService.getEverBeenUnlocked({ userId: userId }));
+        if (neverLock) {
+            // TODO: This also _sets_ the key so when we check memory in the next line it finds a key.
+            // We should refactor here.
```

Comment:
Do we have a tech debt item for this?

Reply:
https://app.asana.com/0/1201461753012676/1201507368222872